www/widgets/DemoExampleMaker.js

```diff
@@ -291,6 +310,47 @@ class DemoExampleMaker extends Widget {
     this.$('[name="dem-url"]').style.display = 'block'
   }
 
+  _uploadJSON (ex) {
+    for (let step = this._data.steps.length; step >= 1; step--) {
+      this._updateStep(step, 'remove')
+    }
+    this._data = {
+      name: ex.name,
+      tags: ex.tags || null,
+      toc: ex.toc || true,
+      layout: ex.layout || 'dock-left',
+      key: Number(ex.key),
+      code: ex.code,
+      steps: ex.info ||
+      [{ title: 'getting started', focus: null, text: 'in this step...' }]
+    }
+    console.log(ex.toc)
+    this.$('[name="dem-demo-name"]').value = ex.name
+    this.$('[name="dem-demo-layout"]').value = ex.layout || 'dock-left'
+    this.$('[name="dem-demo-toc"]').checked = ex.toc || true
+    this.$('[name="dem-demo-tags"]').value = ex.tags || null
+    NNE.code = NNE._decode(ex.code.split('#code/').pop())
+    if (ex.info) {
+      this._data.steps.forEach((step, i) => {
+        if (i === 0 || i < this._data.steps.length) {
```

Comment:
this is looking great @Cthomp7, i've got some notes but b4 i post the full review, could u help me understand why this conditional statement is here?

Reply:
there might be something i'm missing (let me know if i'm over looking a case here), but i'm thinking this should look more like:

```js
    if (ex.info) {
      this._data.steps.forEach((step, i) => {
        this._addStep(step)
        this._selectStep(0)
      })
    } else { 
      this.$('[name="dem-s-title"]').value = null
      this.$('[name="dem-s-focus"]').value = null
      this.$('[name="dem-s-text"]').value = null
    }
```
so that if the example does have "info" (ie. steps) than we add each one && select the first one by default, otherwise we keep the steps array empty && null out the view fields. what do u think?

Reply:
I had line 335 above since before I rewriting over already existing steps and since there always has to be one step I was always updating the preexisting first step. **However** the `steps: ex.info || [{}]` now works for creating the steps in the widget, so I have removed that section of the code in my newest revision.